File: sdk/python/src/capsule_sdk/models/workload.py
from __future__ import annotations

from dataclasses import dataclass

from capsule_sdk.models.common import CapsuleModel


class WorkloadSummary(CapsuleModel):
    """High-level user-facing workload handle."""

    display_name: str
    config_id: str | None = None
    workload_key: str | None = None

    @property
    def name(self) -> str:
        return self.display_name


@dataclass(frozen=True)
class ResolvedWorkloadRef:
    display_name: str | None = None
    config_id: str | None = None
    workload_key: str | None = None

File: sdk/typescript/src/models/workload.ts
export interface WorkloadSummary {
  display_name: string;
  config_id?: string | null;
  workload_key?: string | null;
}

export interface ResolvedWorkloadRef {
  display_name?: string | null;
  config_id?: string | null;
  workload_key?: string | null;
}

export function createResolvedWorkloadRef(
  value: ResolvedWorkloadRef = {},
): ResolvedWorkloadRef {
  return {
    display_name: value.display_name ?? undefined,
    config_id: value.config_id ?? undefined,
    workload_key: value.workload_key ?? undefined,
  };
}

export function createWorkloadSummary(
  value: WorkloadSummary,
): WorkloadSummary {
  return {
    display_name: value.display_name,
    config_id: value.config_id ?? undefined,
    workload_key: value.workload_key ?? undefined,
  };
}

